docs/LoaderApplicationInterface.md

```diff
@@ -0,0 +1,671 @@
+layers and the implementation.
+Thus, using a loader-exported Vulkan device function, the call chain
+in the same scenario as above would look like:
+
+![Loader Device Call Chain](./images/loader_device_chain_loader.png)
+
+An even better solution would be for an application to perform a
+`vkGetDeviceProcAddr` call on all device functions.
+This further optimizes the call chain by removing the loader all-together under
+most scenarios:
+
+![Application Device Call Chain](./images/loader_device_chain_app.png)
+
+Also, notice if no layers are enabled, the application function pointers point
+**directly to the implementation**.
+With many function calls, the lack of indirection in each adds up to non-trivial
+performance savings.
+
+**NOTE:** There are some device functions which still require the loader to
+intercept them with a *trampoline* and *terminator*.
+There are very few of these, but they are typically functions which the loader
+wraps with its own data.
+In those cases, even the device call chain will continue to look like the
+instance call chain.
+One example of a device function requiring a *terminator* is
+`vkCreateSwapchainKHR`.
+For that function, the loader needs to potentially convert the KHR_surface
+object into an implementation-specific KHR_surface object prior to passing down
+the rest of the function's information to the implementation.
+
+Remember:
+ * `vkGetInstanceProcAddr` is used to query instance and physical device
+   functions, but can query all functions.
+ * `vkGetDeviceProcAddr` is only used to query device functions.
+
+
+### ABI Versioning
+
+The Vulkan loader library will be distributed in various ways including Vulkan
+SDKs, OS package distributions and Independent Hardware Vendor (IHV) driver
+packages.
+These details are beyond the scope of this document.
+However, the name and versioning of the Vulkan loader library is specified so
+an app can link to the correct Vulkan ABI library version.
+ABI backwards compatibility is guaranteed for all versions with the same major
+number (e.g. 1.0 and 1.1).
+On Windows, the loader library encodes the ABI version in its name such that
+multiple ABI incompatible versions of the loader can peacefully coexist on a
+given system.
+The Vulkan loader library file name is `vulkan-<ABI version>.dll`.
+For example, for Vulkan version 1.X on Windows the library filename is
+`vulkan-1.dll`.
+This library file can typically be found in the `windows\system32`
+directory (on 64-bit Windows installs, the 32-bit version of the loader with
+the same name can be found in the `windows\sysWOW64` directory).
+
+For Linux and macOS, shared libraries are versioned based on a suffix.
+Thus, the ABI number is not encoded in the base of the library filename as on
+Windows.
+On Linux an application wanting to link to the latest Vulkan ABI version would
+just link to the name vulkan (`libvulkan.so`).  A specific Vulkan ABI version
```

Comment:
In the standard setup for Linux library ABIs, the distinction between `libvulkan.so` and `libvulkan.so.1` on Linux is really about compile-time vs. runtime, rather than latest version vs. a fixed version.

The `.so` symlink is only intended to be used at compile-time: a developer links to libvulkan with something like `cc -lvulkan`, which automatically makes compiler search for `libvulkan.so`.

It would never be safe for an application to load `libvulkan.so` with `dlopen()`, because it could get a Vulkan ABI that is not the one it is expecting, with arbitrary incompatible changes - then it will almost certainly crash. In distributions that separate development libraries from runtime libraries, like Debian and Red Hat, the `.so` symlink is in the `-dev` or `-devel` package and is not installed by end-users, so applications should not assume it will exist.

Reply:
I applied the changes, I'll resolve this and leave the next one open until you've looked over the changes.